pkg/ipcache/metadata.go

```diff
@@ -658,26 +700,28 @@ func (ipc *IPCache) TriggerLabelInjection() {
 
 	// This controller is for retrying this operation in case it fails. It
 	// should eventually succeed.
-	ipc.UpdateController(
-		LabelInjectorName,
-		controller.ControllerParams{
-			Group:   injectLabelsControllerGroup,
-			Context: ipc.Configuration.Context,
-			DoFunc: func(ctx context.Context) error {
-				var err error
-
-				idsToModify := ipc.metadata.dequeuePrefixUpdates()
-				idsToModify, err = ipc.InjectLabels(ctx, idsToModify)
-				ipc.metadata.enqueuePrefixUpdates(idsToModify...)
-
-				return err
-			},
-			MaxRetryInterval: 1 * time.Minute,
-		},
-	)
+	ipc.controllers.TriggerController(LabelInjectorName)
 }
 
-// ShutdownLabelInjection shuts down the controller in TriggerLabelInjection().
-func (ipc *IPCache) ShutdownLabelInjection() error {
-	return ipc.controllers.RemoveControllerAndWait(LabelInjectorName)
+// doLabelInjection is triggered by the ipcache-inject-labels controller
+func (ipc *IPCache) doLabelInjection(ctx context.Context) error {
+	// These tests are duplicated in injectLabels(), but are
+	// also here to prevent unnecessary churning of revisions
+	if ipc.IdentityAllocator == nil {
+		return ErrLocalIdentityAllocatorUninitialized
+	}
+
+	if !ipc.Configuration.CacheStatus.Synchronized() {
+		return errors.New("k8s cache not fully synced")
+	}
+
```

Comment:
Why duplicate them? This is the only functional user.

Reply:
Removed this whole commit.